Mathlib/Analysis/Analytic/Order.lean

```diff
@@ -114,119 +175,177 @@ lemma natCast_le_order_iff (hf : AnalyticAt 𝕜 f z₀) {n : ℕ} :
           ← mul_smul] at hf'
         rw [pow_sub₀ _ (sub_ne_zero_of_ne hz) (by omega), ← hf']
 
+@[deprecated (since := "2025-05-03")] alias natCast_le_order_iff := natCast_le_analyticOrderAt
+
 /-- If two functions agree in a neighborhood of `z₀`, then their orders at `z₀` agree. -/
-theorem order_congr (hf₁ : AnalyticAt 𝕜 f₁ z₀) (h : f₁ =ᶠ[𝓝 z₀] f₂) :
-    (hf₁.congr h).order = hf₁.order := by
-  refine ENat.eq_of_forall_natCast_le_iff fun n ↦ ?_
-  simpa only [natCast_le_order_iff] using ⟨fun ⟨g, hg, hfg⟩ ↦ ⟨g, hg, h.trans hfg⟩,
-    fun ⟨g, hg, hfg⟩ ↦ ⟨g, hg, h.symm.trans hfg⟩⟩
+lemma analyticOrderAt_congr (hfg : f =ᶠ[𝓝 z₀] g) :
+    analyticOrderAt f z₀ = analyticOrderAt g z₀ := by
+  by_cases hf : AnalyticAt 𝕜 f z₀
+  · refine ENat.eq_of_forall_natCast_le_iff fun n ↦ ?_
+    simp only [natCast_le_analyticOrderAt, hf, hf.congr hfg]
+    congr! 3
+    exact hfg.congr_left
+  · rw [analyticOrderAt_of_not_analyticAt hf,
+      analyticOrderAt_of_not_analyticAt fun hg ↦ hf <| hg.congr hfg.symm]
+
+@[deprecated (since := "2025-05-03")] alias AnalyticAt.order_congr := analyticOrderAt_congr
+
+@[simp] lemma analyticOrderAt_neg : analyticOrderAt (-f) z₀ = analyticOrderAt f z₀ := by
+  by_cases hf : AnalyticAt 𝕜 f z₀
+  · refine ENat.eq_of_forall_natCast_le_iff fun n ↦ ?_
+    simp only [ natCast_le_analyticOrderAt, hf, hf.neg]
+    exact (Equiv.neg _).exists_congr <| by simp [neg_eq_iff_eq_neg]
+  · rw [analyticOrderAt_of_not_analyticAt hf,
+      analyticOrderAt_of_not_analyticAt <| analyticAt_neg.not.2 hf]
+
+/-- The order of a sum is at least the minimum of the orders of the summands. -/
+theorem le_analyticOrderAt_add (hf : AnalyticAt 𝕜 f z₀) (hg : AnalyticAt 𝕜 g z₀) :
+    min (analyticOrderAt f z₀) (analyticOrderAt g z₀) ≤ analyticOrderAt (f + g) z₀ := by
+  refine ENat.forall_natCast_le_iff_le.mp fun n ↦ ?_
+  simp only [le_min_iff, natCast_le_analyticOrderAt, hf, hg, hf.add hg]
+  refine fun ⟨⟨F, hF, hF'⟩, ⟨G, hG, hG'⟩⟩ ↦ ⟨F + G, hF.add hG, ?_⟩
+  filter_upwards [hF', hG'] with z using by simp +contextual [mul_add]
+
+@[deprecated (since := "2025-05-03")] alias AnalyticAt.order_add := le_analyticOrderAt_add
+
+lemma le_analyticOrderAt_sub (hf : AnalyticAt 𝕜 f z₀) (hg : AnalyticAt 𝕜 g z₀) :
+    min (analyticOrderAt f z₀) (analyticOrderAt g z₀) ≤ analyticOrderAt (f - g) z₀ := by
+  simpa [sub_eq_add_neg] using le_analyticOrderAt_add hf hg.neg
+
+lemma analyticOrderAt_add_eq_left_of_lt (hf : AnalyticAt 𝕜 f z₀) (hg : AnalyticAt 𝕜 g z₀)
```

Comment:
ditto (note that `hfg` ensures that `g` is analytic, so  `f + g` is analytic iff `f` is)